types/draco3d/index.d.ts

```diff
@@ -0,0 +1,197 @@
+    Encoder: new () => Encoder;
+    MeshBuilder: new () => MeshBuilder;
+
+    POSITION: number;
+    NORMAL: number;
+    TEX_COORD: number;
+    COLOR: number;
+    GENERIC: number;
+
+    MESH_SEQUENTIAL_ENCODING: number;
+    MESH_EDGEBREAKER_ENCODING: number;
+}
+
+export interface DecoderModule extends BaseModule {
+    Decoder: new () => Decoder;
+    DecoderBuffer: new () => DecoderBuffer;
+
+    // GeometryType.
+    TRIANGULAR_MESH: GeometryType;
+    POINT_CLOUD: GeometryType;
+
+    // DataType.
+    DT_FLOAT32: DataType;
+    DT_INT8: DataType;
+    DT_INT16: DataType;
+    DT_INT32: DataType;
+    DT_UINT8: DataType;
+    DT_UINT16: DataType;
+    DT_UINT32: DataType;
+}
+
+export interface Encoder {
+    SetAttributeQuantization(attribute: number, bits: number): void;
+    SetAttributeExplicitQuantization(
+        attribute: number,
+        bits: number,
+        itemSize: number,
+        origin: [number, number, number],
+        range: number
+    ): void;
+    SetSpeedOptions(encodeSpeed: number, decodeSpeed: number): void;
+    SetEncodingMethod(method: number): void;
+    SetTrackEncodedProperties(track: boolean): void;
+    EncodeMeshToDracoBuffer(mesh: Mesh, array: DracoInt8Array): number;
+    GetNumberOfEncodedPoints(): number;
+    GetNumberOfEncodedFaces(): number;
+}
+
+export interface Decoder {
+    DecodeBufferToMesh(buffer: DecoderBuffer, mesh: Mesh): Status;
+    GetAttributeByUniqueId: (mesh: Mesh, id: number) => Attribute;
+    GetFaceFromMesh: (mesh: Mesh, index: number, array: DracoArray) => number;
+    GetTrianglesUInt16Array: (mesh: Mesh, byteLength: number, ptr: number) => void;
+    GetTrianglesUInt32Array: (mesh: Mesh, byteLength: number, ptr: number) => void;
+    GetAttributeDataArrayForAllPoints: (
+        mesh: Mesh,
+        attribute: Attribute,
+        type: DataType,
+        byteLength: number,
+        ptr: number
+    ) => void;
+    GetAttributeFloatForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
+    GetAttributeInt8ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
+    GetAttributeInt16ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
+    GetAttributeInt32ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
+    GetAttributeUInt8ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
+    GetAttributeUInt16ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
+    GetAttributeUInt32ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
+    GetEncodedGeometryType: (buffer: DecoderBuffer) => GeometryType;
+}
+
+export interface DecoderBuffer {
+    Init: (array: Int8Array, byteLength: number) => void;
+}
+
+export interface DracoArray {
+    GetValue: (index: number) => number;
+}
+
+// tslint:disable-next-line:no-empty-interface
+export interface DracoFloat32Array extends DracoArray {}
```

Comment:
Is there a reason you need to use `interface`? Couldn't `export type DracoFloat32Array = DracoArray` work for the below declarations?

Reply:
I think the downside of this would be that methods like `EncodeMeshToDracoBuffer` (not sure if there are others?) will not be able to typecheck that the input array is of the correct type, if they're all aliases of `DracoArray`. I'm fine with either though.